### DISCUSSION TITLE
제페토 패키지 빌드 파일 용량 관련 문의

### DISCUSSION THREAD

**Discussion by kihye** [naverz/zepeto-studio-kor#566] [5. Q&A]
![image](https://user-images.githubusercontent.com/80503690/161197748-e3bcee81-2e89-463d-9978-0c58580e5af6.png)
이렇게 옵션 주고 import size대로 패키징되는 줄 알았는데 로그 열어보니까

![image](https://user-images.githubusercontent.com/80503690/161197872-4497b9b0-0c39-4cb0-bf49-37e4a7840080.png)
기존 사이즈 그대로 들어가버리네요..
그 외 이미지 등 다른 파일도 유니티 인스펙터에서 뜨는 파일 크기와는 다른 크기(기존 파일 크기)로 빌드 로그에 찍힙니다.

제페토 패키지 용량을 줄이기 위해서 유니티에서 제공하는 압축 옵션을 적용시켜봤는데 제페토 패키지 파일 빌드할 땐 이런 압축 옵션이 아예 적용이 안 되는게 맞는건가요? 아니라면 따로 적용시킬만한 방법이 있을까요?

이것저것 다 건드려보고도 도무지 더 방법을 찾을 수가 없어서 이렇게 글 올려봅니다...

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, zepetopackage 파일에는 `압축되어 런타임에 사용될 빌드`와 함께 `심사 및 이슈 추적을 위한 원본 리소스`도 같이 패키징이 되게 됩니다.
그래서 원본 용량도 중요하며, 또한 압축된 용량 모두 최종 결과 파일에 영향을 미치게 됩니다.
참고로 실제 플레이하는 유저들은 압축된 용량의 데이터만 사용이 되니 참고 부탁드립니다

**Reply by kihye** [reply]
다 고려해도 패키지파일 자체를 500메가 이하로 맞춰서 심사를 봐야된다는 점은 변함없나요?

**Reply by Zepeto-TechEvangelist** [reply]
넵 맞습니다.

**Reply by kihye** [reply]
아무리 그래도 레거시 버전보다 달라진 부분이 그렇게 크지 않은데도 용량 차이가 너무 많이 나는 이유를 로그를 봐도 알 수가 없어서 그렇습니다... 혹시 조금이나마 도움이 될만한 방법은 없을까요?

**Reply by kihye** [reply]
특히나 windows 파일이 60메가인데 비해 android 파일이 250메가나 되는 현상을 아무리 원인을 알아보려 해도 알 수가 없어서 답답한 마음입니다..

**Reply by Zepeto-TechEvangelist** [reply]
Project Settings > Quality 에서 Windows와 Android 퀄리티 차이가 많이 나는지 확인 부탁드립니다.

**Reply by kihye** [reply]
Windos가 Ultra, ios와 Android가 High입니다

**Reply by kihye** [reply]
옵선을 다 Very Low로 바꾸고 리빌드해도 별로 유의미한 차이는 없는 것 같네요..용량은 같습니다

**Reply by Zepeto-TechEvangelist** [reply]
정확한 확인을 해보려면 프로젝트 파일을 봐야 할 것 같습니다.

**Reply by kihye** [reply]
업무 관련이라 프로젝트 파일 전체를 공유해드리기는 좀 어려울 것 같습니다...ㅜㅜ

**Reply by cheolhui-lee** [reply]
공유가 어려우시다면, 프로젝트에 포함된 리소스들을 직접 Asset Bundle 로 빌드해보시면서 확인하실 수 있을 것 같습니다.
https://docs.unity3d.com/Manual/AssetBundles-Building.html

대략 추정되는 부분은 텍스쳐나 오디오 파일들의 압축 효율이 안좋거나 할 수 있어보입니다.
압축은 OS에 따라 다른 형태로 진행이 되거든요.

**Reply by kihye** [reply]
올려주신 링크 참고해보겠습니다!

**Reply by kihye** [reply]
원인은 잘 모르겠지만 아마 압축 관련으로 문제가 있었던 것 같습니다.
제가 사용한 방법으로는 일단 android 자체 어플리케이션 빌드 후 제페토 빌드 할때 android 파일 용량이 줄어든 걸 확인했고,
ios도 마찬가지로 ios 빌드 후(windows 환경이라 끝나고 아웃풋은 안나왔습니다.) 제페토 빌드를 다시 돌려보니 ios 파일 용량이 줄어들었습니다.
둘 다 Compresson Method는 LZ4HC를 사용했으며 최종적으로 패키지파일을 879mb->306mb까지 줄이는 데 성공했습니다.

**Reply by cheolhui-lee** [reply]
이슈가 해결이 되었다면 다행입니다.
다른 분들도 참고하실 수 있게 간략히 상황 공유 부탁드려도 될까요?

**Reply by kihye** [reply]
android 파일과 ios 파일 둘 다 모두 260mb를 웃돌았었는데 이 방법을 통해 못해도 70mb 아래로 내려갔네요

**Reply by kihye** [reply]
제 경우에는 프로젝트를 진행하며 제페토 빌드를 여러번 돌리곤 했었는데, 그때까진 모바일 관련 빌드는 전혀 하지 않았고 패키징 된 파일 용량만 보려고 windows 빌드만 했었던 걸로 기억합니다.

**Reply by kihye** [reply]
원래는 거의 제페토 패키지 파일이 2기가 가까이 됐던걸로 기억하는데 텍스쳐랑 이것저것 파일 최적화 하고도 500메가를 훌쩍 넘겨버리는 상황이었습니다

**Reply by kihye** [reply]
아마 이 방법이 다른 분들에게도 유효할지는 모르겠지만 제페토 패키지 파일 용량이 이상하게 클 경우에 참고하면 좋을 것 같네요
(특히 제페토 빌드 후 android, ios 파일 용량이 이상하게 클 때)
